### PR TITLE
Upgrade qbits backend

### PR DESCRIPTION
**Pull request by zhewang1-intc**
## Type of Change

feature or bug fix or documentation or others: feature
API changed or not: no

## Description
enable avx_vnni ISA in qbits & fix windows compile

## Expected Behavior & Potential Risk

the expected behavior that triggered by this PR 

## How has this PR been tested?

how to reproduce the test (including hardware information)

## Dependency Change?

any library dependency introduced or removed

### CODE REVIEW

**Comment by VincyZhang** [comment]
windows build https://inteltf-jenk.sh.intel.com/job/nlp-release-build-win/268/

File: intel_extension_for_transformers/llm/operator/cscr/dispatcher/src/jblas_weightonly_dispatcher.cpp [hunk 1]
```diff
@@ -491,19 +522,24 @@ void parse_gemm_core_offline(qbits_config_param* p, qbits_runtime_ctx* ctx) {
...
+      TORCH_CHECK(false, "Qbits: device ISA must lagger than AVX512_VNNI when GemmCore==Row_NN_8x48_AVX512_VNNI");
+      break;
+    case jblas::gemm::GemmCoreType::AVX_VNNI_2x48:
+      assert(p->compute_type == "int8");
+      if (check_avx_vnni())
+        return parse_weight<TASK, jblas::wrapper::gemm_pack_weight::GemmInterfaceParallelAB,
+                            jblas::wrapper::gemm_pack_weight::GemmLauncherPackWeight,
+                            jblas::gemm::GemmCore_Row_NN_2x48_AVX_VNNI, jblas::utils::parallel::Parallel2DGemm,
+                            JblasAVX_VNNI>(p, ctx);
+      TORCH_CHECK(false, "Qbits: device ISA must lagger than AVX_VNNI when GemmCore==Row_NN_2x48_AVX_VNNI");
```
Comment (airMeng) [inline intel_extension_for_transformers/llm/operator/cscr/dispatcher/src/jblas_weightonly_dispatcher.cpp:534]: ```large``` is an internal definition.
```suggestion
      TORCH_CHECK(false, "Qbits: device ISA must support AVX_VNNI when GemmCore==Row_NN_2x48_AVX_VNNI");
```

File: intel_extension_for_transformers/llm/operator/cscr/dispatcher/src/jblas_weightonly_dispatcher.cpp [hunk 2]
```diff
@@ -491,19 +522,24 @@ void parse_gemm_core_offline(qbits_config_param* p, qbits_runtime_ctx* ctx) {
...
     case jblas::gemm::GemmCoreType::AMX_INT8_16x48:
       if (check_amx())
         return parse_weight<TASK, jblas::wrapper::gemm_pack_weight::GemmInterfaceParallelAB,
                             jblas::wrapper::gemm_pack_weight::GemmLauncherPackWeight,
                             jblas::gemm::GemmCore_Row_NN_16x48_AMX_S8S8, jblas::utils::parallel::Parallel2DGemm,
                             JblasAMX_INT8>(p, ctx);
-      TORCH_CHECK(false,
-                  "Qbits: device ISA must lagger than AMX_INT8 when "
-                  "GemmCore==Row_NN_16x48_AMX_S8S8");
+      TORCH_CHECK(false, "Qbits: device ISA must lagger than AMX_INT8 when GemmCore==Row_NN_16x48_AMX_S8S8");
```
Comment (airMeng) [inline intel_extension_for_transformers/llm/operator/cscr/dispatcher/src/jblas_weightonly_dispatcher.cpp:542]:
```suggestion
      TORCH_CHECK(false, "Qbits: device ISA must support AMX_INT8 when GemmCore==Row_NN_16x48_AMX_S8S8");
```

**Review by airMeng — APPROVED** [review]

File: intel_extension_for_transformers/llm/library/jblas/jblas/kernel_avx2.h [hunk 3]
```diff
@@ -50,6 +50,10 @@ static inline __m128i unpack_4bits_sse(void* srcptr) {
 template <JBLAS_SIGN_INT_TYPE S4_T>
 static inline void convert_s4_s8_16_sse(int8_t* dstptr, int8_t* srcptr) {
   auto dst0 = unpack_4bits_sse<S4_T>(srcptr);
+  if constexpr (S4_T == S4_FULLRANGE) {
```
Comment (hshen14) [inline intel_extension_for_transformers/llm/library/jblas/jblas/kernel_avx2.h:53]: do we have some results to show the acc benefits of full range? @xin3he  @zhewang1-intc

File: intel_extension_for_transformers/llm/operator/cscr/dispatcher/src/jblas_weightonly_dispatcher.cpp (hunk 2 above)
Comment (zhewang1-intc) [reply intel_extension_for_transformers/llm/operator/cscr/dispatcher/src/jblas_weightonly_dispatcher.cpp]: fix in [ca30dcc](https://github.com/intel/intel-extension-for-transformers/pull/460/commits/ca30dccf641ecff8c2b74cc545c9b44e2d3fc96b)

**Pull request merged**